core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -108,30 +118,109 @@ final case class OptionT[F[_], A](value: F[Option[A]]) {
   def getOrElseF[B >: A](default: => F[B])(implicit F: Monad[F]): F[B] =
     F.flatMap(value)(_.fold(default)(F.pure))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT(List(Some(2), None, Some(414), None, None))
+   *  scala> optionT.collect{ case i: Int => i == 2 }
```

Comment:
I think this example would be better if you did `case i if i == 2 => i `

Because the way you wrote it this is the same as a map function?

Reply:
Just like @FelAl's is a glorified `map`, @johnynek's is a glorified `filter`.  Doing an inexhaustive match _and_ doing something other than returning `i` on the right would best show both those characteristics of `collect`.

Reply:
I've provided both examples.